source/ship/ShipAICache.cpp

```diff
@@ -0,0 +1,117 @@
+/* ShipAICache.cpp
+Copyright (c) 2022 by Hurleveur
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "ShipAICache.h"
+
+#include "../pi.h"
+#include "../Ship.h"
+
+#include <algorithm>
+#include <cmath>
+
+using namespace std;
+
+
+
+// Constructor.
```

Comment:
nice info thanks 😆

Reply:
*Crucial* info!

Reply:
```suggestion
```
I'm not the one who wrote this xD

Reply:
you dont have to remove it, just found it funny

Reply:
we need more comments, maybe not like this though...